### ISSUE TITLE
[chart] Actions for Legend

### ISSUE THREAD

**Issue by holospice18** [mui/mui-x#12344]
### Search keywords

chart

### Latest version

- [X] I have tested the latest version

### Summary

Is it possible to have events on legends like highlight when hover on corresponding legend?

### Examples

Chartjs has action events for legends
https://www.chartjs.org/docs/latest/samples/legend/events.html

Maybe "hilghlight" is enough for now and not needed the filtering

### Motivation

To have a shortcut navigation to highlight item or series, especially for large data

**Comment by michelengelen** [comment]
Hey @holospice18 that is a very neat feature, but unfortunately currently not supported in our charts components.

I did add this to our board, so thanks for raising this feature request.

cc @alexfauquette

**Comment by TheOneTheOnlyJJ** [comment]
This would be great to have. Default dataset filtering through the legend would make a very good addition to data science applications.

**Comment by trungutt** [comment]
Agree, I think it should be the default behavior of legend in case of pie/donut charts. Or I'd love to know if we have a "manual" mechanism to highlight item (knowing what legend is hovering and set the highlight on the corresponding item of the pie)

**Comment by MartinKallinger** [comment]
I am not sure If that is already covered by the filtering idea. But what would be nice for our application, would be a possibility to hide certain values (bars). For example, in a bar chart with two bars(wins, losses) per year, it would be nice to hide the "losses" bar through the legend, so that only the wins per year are shown.

**Comment by sterlingdcs-damian** [comment]
I've been spending the past few hours trying to find out how i can hook into an `onClick` event for a legend item so i can hide parts of the chart. Seems like this is not possible and i'd need to write my own legend component?
From what i can ascertain there are only onItemClick events for the bars themselves but not for the legend squares.

**Comment by JCQuintas** [comment]
Hi @sterlingdcs-damian, that is correct

**Comment by prakhargupta1** [comment]
I found an implementation where hovering over the pie slice also highlights the legend entry. Let’s consider supporting both interactions for better UX.

https://github.com/user-attachments/assets/f3b2439e-5300-4655-a438-15f11d4ca297
